test/test_nn.py

```diff
@@ -1118,6 +1118,25 @@ def names(named_buffers):
             names(s.named_buffers()),
             ['0.dummy_buf', '0.l1.layer_dummy_buf'])
 
+    def test_named_parameters_buffers_duplicates(self):
+        class Foo(nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.bias = nn.Parameter(torch.randn(3))
+                self.linear = nn.Linear(3, 3)
+                self.linear.bias = self.bias
+                self.linear_cloned = self.linear
+                self.register_buffer('buffer', torch.randn(3))
+                self.linear.buffer_cloned = self.buffer
```

Comment:
I think `self.register_buffer('buffer_cloned', self.buffer)` works for cloning a buffer